github/github.go

```diff
@@ -323,7 +324,7 @@ func parseRate(r *http.Response) Rate {
 // current rate.
 func (c *Client) Rate() Rate {
 	c.rateMu.Lock()
-	rate := c.rate
+	rate := c.rateLimits[c.mostRecent]
```

Comment:
actually yeah, we don't really need to provide another mechanism... we already instruct users to call client.RateLimits for the most up to date values.  That should be sufficient, no?